SimTKcommon/include/SimTKcommon/internal/ParallelExecutor.h

```diff
@@ -80,13 +89,22 @@ class SimTK_SimTKCOMMON_EXPORT ParallelExecutor : public PIMPLHandle<ParallelExe
      */
     void execute(Task& task, int times);
     /**
-     * Get the number of available processor cores.
+     * Get the total number of available processor cores.
      */
     static int getNumProcessors();
     /**
+      * Get the number of physical processor cores.
+      */
+    static int getNumPhysicalProcessors();
+    /**
      * Determine whether the thread invoking this method is a worker thread created by ParallelExecutor.
      */
     static bool isWorkerThread();
+private:
+    /**
+     * Maximum number of threads the ParallelExecutor is allowed to use
+     */
+    int NUM_MAX_THREADS; //TODO: Bad style?
```

Comment:
Is it really necessary for both `ParallelExecutor` and `ParallelExecutorImpl` to have this member variable?